common/src/app/common/pages/changes_builder.cljc

```diff
@@ -276,6 +276,21 @@
          (update :undo-changes #(reduce mk-undo-change % shapes))
          (apply-changes-local)))))
 
+
+(defn changed-attrs
+  [object update-fn {:keys [attrs]}]
+  (let [changed?
+        (fn [old new attr]
+          (let [old-val (get old attr)
+                new-val (get new attr)]
+            (not= old-val new-val)))
+        new-obj (update-fn object)]
+    (if (= object new-obj)
+      '()
```

Comment:
Can be simple when, returning `nil` in some cases, `nil` and `'()` are equivalent in seq processing

Reply:
ok